### ISSUE TITLE
CronJob produces seemingly random warnings for sandbox creation failures

### ISSUE THREAD

**Issue by blitzd** [kubernetes/kubernetes#95916]
**What happened**:
CronJob produces seemingly random warnings relating to sandbox creation issues

**What you expected to happen**:
No random warnings

**How to reproduce it (as minimally and precisely as possible)**:
Deploy the sample cronjob, within 10 runs on a 1-minute schedule you should see a warning about sandbox creation. 

**Anything else we need to know?**:
Does not appear to be related to memory pressure - or memory pressure is not required for it to occur, but may impact frequency. There is a corresponding issue in the docker repository here https://github.com/moby/moby/issues/40835

**Environment**:
- Kubernetes version (use `kubectl version`):
1.16 EKS, 1.17 EKS, 1.19.2 (Docker Desktop)

- Cloud provider or hardware configuration:
AWS/EKS

- OS (e.g: `cat /etc/os-release`):
Amazon Linux 2

- Errors:

``
Failed create pod sandbox: rpc error: code = Unknown desc = failed to start sandbox container for pod "hello-1603208700-v642l": Error response from daemon: OCI runtime create failed: container_linux.go:349: starting container process caused "process_linux.go:319: getting the final child's pid from pipe caused \"EOF\"": unknown
``

``
Failed create pod sandbox: rpc error: code = Unknown desc = failed to start sandbox container for pod "hello-1603207560-kd9vt": Error response from daemon: OCI runtime create failed: container_linux.go:349: starting container process caused "process_linux.go:449: container init caused \"read init-p: connection reset by peer\"": unknown
``

``
Failed to create pod sandbox: rpc error: code = Unknown desc = failed to start sandbox container for pod "hello-1603214100-l6pvl": Error response from daemon: OCI runtime create failed: container_linux.go:349: starting container process caused "process_linux.go:449: container init caused \"process_linux.go:432: running prestart hook 0 caused \\\"error running hook: exit status 1, stdout: , stderr: time=\\\\\\\"2020-10-20T17:15:06Z\\\\\\\" level=fatal msg=\\\\\\\"no such file or directory\\\\\\\"\\\\n\\\"\"": unknown
``

``
Failed to create pod sandbox: rpc error: code = Unknown desc = failed to start sandbox container for pod "hello-1603219200-5tl69": Error response from daemon: OCI runtime create failed: container_linux.go:349: starting container process caused "process_linux.go:449: container init caused \"\"": unknown
``

``
Failed create pod sandbox: rpc error: code = Unknown desc = failed to start sandbox container for pod "hello-1603468740-5wq58": Error response from daemon: OCI runtime create failed: container_linux.go:349: starting container process caused "process_linux.go:449: container init caused \"process_linux.go:438: writing syncT 'resume' caused \\\"write init-p: broken pipe\\\"\"": unknown
``

**Comment by blitzd** [comment]
An additional four error variations of the error that we have since received:

```
Failed to create pod sandbox: rpc error: code = Unknown desc = failed to set up sandbox container "86d6b0bca11814d7b1553df3dd96bfe88ea018116b540e11f9b4f7b23715a26f" network for pod "hello-1605116460-85tdk": networkPlugin cni failed to set up pod "hello-1605116460-85tdk_default" network: add command: failed to setup network: setupNS network: failed to setup NS network: failed to Statfs "/proc/12364/ns/net": no such file or directory

Failed to create pod sandbox: rpc error: code = Unknown desc = failed to start sandbox container for pod "hello-1605126000-6ddmw": Error response from daemon: OCI runtime start failed: cannot start a container that has stopped: unknown

Failed to create pod sandbox: rpc error: code = Unknown desc = failed to start sandbox container for pod "hello-1605125160-8s4xx": Error response from daemon: OCI runtime create failed: container_linux.go:349: starting container process caused "process_linux.go:327: getting pipe fds for pid 32114 caused \"readlink /proc/32114/fd/0: no such file or directory\"": unknown

Failed to create pod sandbox: rpc error: code = Unknown desc = failed to start sandbox container for pod "hello-1605123960-2b475": Error response from daemon: OCI runtime create failed: container_linux.go:349: starting container process caused "process_linux.go:449: container init caused \"process_linux.go:432: running prestart hook 0 caused \\\"error running hook: exit status 1, stdout: , stderr: time=\\\\\\\"2020-11-11T19:46:08Z\\\\\\\" level=fatal msg=\\\\\\\"no such file or directory\\\\\\\"\\\\n\\\"\"": unknown
```

**Comment by blitzd** [comment]
Another variation:

``
Failed to create pod sandbox: rpc error: code = Unknown desc = failed to set up sandbox container "623a315b43c55240719c14a0d963cf8ebe8649209cc5d0e12ff9f611146af4e8" network for pod "hello-1605207300-dbx89": networkPlugin cni failed to set up pod "hello-1605207300-dbx89_default" network: failed to Statfs "/proc/26303/ns/net": no such file or directory
``

**Comment by ehashman** [comment]
@blitzd can you please provide a copy of the full logs from the node/kubelet?

/priority waiting-additional-evidence

**Comment by ehashman** [comment]
sorry I can't spell

/priority awaiting-more-evidence

**Comment by blitzd** [comment]
@ehashman We opted to not use CronJob because of these issues, and I am no longer in a position where I can spend much time on it. The problem was easily re-created on any system that I attempted it on though, both by myself and AWS support.

**Comment by ehashman** [comment]
@blitzd I'm guessing this is an instance of a race condition in the kubelet runtime (similar to #98142 but using containerd) but it's impossible to say for sure without logs. I don't have access to EKS clusters to attempt to reproduce this.

If anyone could provide full kubelet logs (v=4 would be great) from when this issue occurs, we should be able to say with more certainty.

**Comment by blitzd** [comment]
@ehashman this was not specific to EKS, I had no issues re-creating it with Kubernetes on docker-desktop. I've attached a slice of the kubelet log from that.

[CronJob-kubelet.log](https://github.com/kubernetes/kubernetes/files/5877309/CronJob-kubelet.log)

**Comment by ehashman** [comment]
/triage more-info

@blitzd can you confirm if https://github.com/kubernetes/kubernetes/pull/98424/ fixed the issue? This landed in any 1.21.x or 1.22.x release (not eligible for backport).

**Comment by blitzd** [comment]
> /triage more-info
> 
> @blitzd can you confirm if #98424 fixed the issue? This landed in any 1.21.x or 1.22.x release (not eligible for backport).

@ehashman I was able to test this finally and I did not run into any warnings or errors after running the sample cronjob from the k8s docs for 24 hours.

**Comment by ehashman** [comment]
Based on https://github.com/kubernetes/kubernetes/issues/95916#issuecomment-877313463 I will mark this as closed.

Thanks :partying_face: 

/remove-lifecycle rotten
/close
